api/src/main/java/jakarta/data/page/Slice.java

```diff
@@ -74,6 +74,14 @@ public interface Slice<T> extends Streamable<T> {
      */
     int numberOfElements();
 
+    /**
+     * Returns {@code true} if it is known that there are more results or that it is
+     * necessary to request a next page to determine whether there are more results, so that
+     * {@link #nextPageRequest()} will definitely not return {@code null}.
+     * @return {@code false} if this is the last page of results.
+     */
+    boolean hasNext();
```

Comment:
The same point here.

Include the UnsupportedOperationException if the database does not know.

Reply:
No -that wrecks the whole pattern.  The point is that if the provider does not know if there is going to be a next page, then we want the user to request it to find out and the user should be able to do that without having to experience exceptions.

Reply:
@otaviojava I think the idea here is that if you have a full last page of results you might not know that it is the last page. What @njr-11 is trying to permit here is that there be an "empty last page". Which I think is fine and correct. Providers can in principle use a "fancy" implementation that avoids this, but I don't think we should mandate such fanciness.